sdk/core/core-amqp/test/requestResponse.spec.ts

```diff
@@ -86,19 +86,6 @@ describe("RequestResponseLink", function() {
     const request: AmqpMessage = {
       body: "Hello World!!"
     };
-    setTimeout(() => {
-      rcvr.emit("message", {
-        message: {
-          correlation_id: messageId,
-          application_properties: {
-            statusCode: 500,
-            errorCondition: ErrorNameConditionMapper.InternalServerError,
-            statusDescription: "Please retry later.",
-            "com.microsoft:tracking-id": "1"
-          }
-        }
-      });
-    }, 2000);
```

Comment:
With this change, there is essentially no difference between what is being tested in this test case and the one before this.

The test case here mocks an initial `InternalServerError`, then tests that a retry attempt is made which is then mocked to be successful.

Instead of removing the mocking of the retryable error, we can replace the `await link.sendRequest(...)` call with a `retry()` where operation would be `link.sendRequest`

Reply:
Ah got it, since retry is no longer part of `RequestResponse` sendRequest and moved out to the individual operations in SDKs. Wouldn't be better to have "successful request response" test case be added in Event Hubs?
`retry`in `core-amqp` already has tests for retry correct? It seems to make sense to remove this test because otherwise it seems like we are just testing the `retry` itself, because the `InternalServerError` is not real and just a mocked value.

Reply:
The benefit of having a test case that mocks an error from the service is that we ensure that such an error is getting bubbled up as expected. 
Having retry here ensures that a subsequent attempt on the same link is successful after a prior error.

So, I still see value in it.
I'd hate to see a nice test with all the mocking go to waste just because we moved the retry logic to Event Hubs.
Especially because trying to have a similar test in Event Hubs will take more time (because we dont have the precedence of mocking a connection/session/link there) than tweaking what we have here.

Reply:
I see the required updates have been done to this test case along with being renamed to `should surface error up through retry`. But, since it got moved to the end of the file, it is a little hard to see the actual changes that were made to the test.

Please bring the test case back up so that it is the second from top. With that, the diff will show exactly what changed and will help in reviewing

Reply:
Thanks! Looks great now